src/Stack/Hoogle.hs

```diff
@@ -100,6 +100,17 @@ hoogleCmd (args,setup,rebuild) go = withBuildConfig go pathToHaddocks
         hooglePackageIdentifier <-
             do (_,_,resolved) <-
                    resolvePackagesAllowMissing
+
+                       -- FIXME this Nothing means "do not follow any
+                       -- specific snapshot", which matches old
+                       -- behavior. However, since introducing the
+                       -- logic to pin a name to a package in a
+                       -- snapshot, we may arguably want to ensure
+                       -- that we're grabbing the version of Hoogle
+                       -- present in the snapshot currently being
+                       -- used.
```

Comment:
The behaviour described in the FIXME does seem more expected to me.  I wonder how many people rely on the current behaviour.